packages/expo-doctor/src/checks/StoreCompatibilityCheck.ts

```diff
@@ -0,0 +1,66 @@
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+import { learnMore } from '../utils/TerminalLink';
+import { existsAndIsNotIgnoredAsync } from '../utils/files';
+
+export class StoreCompatibilityCheck implements DoctorCheck {
+  description = 'Check if the project meets version requirements for submission to app stores';
+
+  sdkVersionRange = '*';
+
+  async runAsync({ exp, projectRoot }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    let issue: string | undefined = undefined;
+
+    // *** non-CNG ***
+    if (await existsAndIsNotIgnoredAsync(path.join(projectRoot, 'android', 'build.gradle'))) {
+      const buildGradle = fs.readFileSync(
+        path.join(projectRoot, 'android', 'build.gradle'),
+        'utf8'
+      );
+      // checks for exact string from SDK 49 bare template, will not work if it was changed
+      const targetSdkVersionRegex =
+        /^\s*targetSdkVersion\s*=\s*(?:Integer\.parseInt\(findProperty\('android\.targetSdkVersion'\)\s*\?:\s*'(\d+)'\)|'(\d+)')/m;
+      const match = buildGradle.match(targetSdkVersionRegex);
+      const targetSdkVersion = match ? parseInt(match[1], 10) : undefined;
+      if (targetSdkVersion && targetSdkVersion < 34) {
+        issue = 'This project appears to be targeting Android API level 33 or lower. ';
+      }
+    } else {
+      // *** CNG ***
+
+      // check if build properties overrides to a lower target SDK
+      const buildPropertiesConfig = exp.plugins?.find(
+        (plugin) => plugin[0] === 'expo-build-properties'
+      );
+      if (
+        buildPropertiesConfig &&
+        buildPropertiesConfig.length > 1 &&
+        buildPropertiesConfig[1].android.targetSdkVersion < 34
+      ) {
+        issue =
+          'This project is using expo-build-properties to target Android API level 33 or lower. ';
+      } else if (!semver.satisfies(exp.sdkVersion!, '>=50.0.0')) {
```

Comment:
We also could use a constant here, or maybe even lookup the template for the current version and determine it through the template files. That way, it auto-updates itself.